docs/proposals/kubemark.md

```diff
@@ -0,0 +1,164 @@
+will appear in Kubemark results from either CPU/Memory preassure from other things running on Nodes (e.g. FluentD, or Kubelet) or running
+cluster over an overlay network. We belive that it'll be possible to turn off cluster monitoring for Kubemark runs, so that the impact
+of real Node daemons will be minimized, but we don't know what will be the impact of using higher level networking stack. Running a
+compariston will be an interesting test in itself.
+
+To make Kubemark easier to maintain when system evolves Hollow components will reuse real "production" code for Kubelet and KubeProxy, but will mock all the backends with no-op or very simple mocks. We belive that this approach is better in the long run than writing special "performance-test-aimed" separate version of them. This may take more time to create an initial version, but we think maintenance cost will be noticeably smaller.
+
+![Kubmark architecture diagram](Kubemark_architecture.png?raw=true "Kubemark architecture overview")
+
+## Things to simulate
+
+In real Kubernetes on a single Node we run two daemons that communicate with Master in some way: Kubelet and KubeProxy. In this paragraph we list what kind of communication we
+have between those daemons and Master components. First step in creating holograms is understanding the communication between Master and Node components.
+
+### KubeProxy
+
+Creates two Watch clients, one for Services, one for Endpoints (it’s done in pkg/proxy/config/api.go, because why not). KubeProxy also exposes healthz, but it’s not read by
+the Master (or anything else except supervisord AFAICT). As a replacement for KubeProxy we'll use HollowProxy, which will be a real
+KubeProxy with injected no-op mocks everywhere it makes sense.
+
+### Kubelet
+
+More complex of the two. Kubelet watches Services, and Nodes (in separate Watch clients, created in pkg/kubelet/kubelet.go) and Pods in another client (created in
+pkg/kubelet/config/apiserver.go because it’s the most obvious place to put it). On top of reading data from API server using those 3 Watch clients Kubelet pushes updates of
+PodStatuses and NodeStatuses using appropriate subresources. As a replacement for Kubelet we'll use HollowKubelet, which will be a real
+Kubelet with injected no-op or simple mocks everywhere it makes sense.
+
+In addition Kubelet exposes cadvisor endpoint which is scraped by Heapster, healthz to be read by supervisord, and we have FluentD running as a Pod on each Node that exports
+logs to Elasticsearch (or Google Cloud Logging). Both Heapster and Elasticsearch are running in Pods in the cluster so do not add any load on a Master components by
+themselves. There can be other systems that scrape Heapster through proxy running on Master, which adds additional load, but they're not the part of default setup
+
+This means that the Hollow version of Kubelet will have 3 Watch clients, and regularly push Status updates to the Master. In the first version we’ll assume that all started
```

Comment:
The first sentence here doesn't provide much value - it will be real Kubelet, so we don't need to change anything here.
I would just leave the second sentence: "In the first version we'll assume..."